tests/unit/RegistryTestCase.php

```diff
@@ -15,66 +15,162 @@ public function setUp(): void {
 		$settings['public_introspection_enabled'] = 'on';
 		update_option( 'graphql_general_settings', $settings );
 
+		\WPGraphQL::clear_schema();
 		$type_registry  = \WPGraphQL::get_type_registry();
 		$this->instance = new Registry( $type_registry, \WP_Block_Type_Registry::get_instance() );
 	}
 
-	/**
-	 * @covers Registry->load_registered_editor_blocks
-	 */
-	public function test_load_registered_editor_blocks_callback() {
-		global $wp_filter;
-		$this->assertTrue( isset( $wp_filter['graphql_register_types']->callbacks ) );
-		$config = $this->instance->load_registered_editor_blocks( array() );
-		$this->assertEquals( $config['registered_editor_blocks'], $this->instance->registered_blocks );
+	public function tearDown(): void {
+		// your tear down methods here
+		\WPGraphQL::clear_schema();
+		parent::tearDown();
 	}
 
 	/**
-	 * @covers Registry->get_supported_post_types
+	 * This test ensures that the `register_interface_types()` method
+	 * works as expected when no get_allowed_block_types is used
 	 */
-	public function test_get_supported_post_types() {
-		$expected_post_types = array(
-			'Post',
-			'Page',
+	public function test_add_block_fields_to_schema_no_get_allowed_block_types() {
+		$query = '
+		query GetType($name:String!) {
+			__type(name: $name) {
+				interfaces {
+					name
+                    description
+				}
+			}
+		}
+		';
+
+		$this->instance->OnInit();
+
+		// Verify the response contains what we put in cache
+		$response           = graphql(
+			array(
+				'query'     => $query,
+				'variables' => array(
+					'name' => 'Post',
+				),
+			)
+		);
+		$contains_interface = array(
+			'name'        => 'NodeWithEditorBlocks',
+			'description' => 'Node that has content blocks associated with it',
 		);
-		$this->assertEquals( $this->instance->get_supported_post_types(), $expected_post_types );
+		$this->assertArrayHasKey( 'data', $response, json_encode( $response ) );
+		$this->assertNotEmpty( $response['data']['__type']['interfaces'] );
+		$this->assertTrue( in_array( $contains_interface, $response['data']['__type']['interfaces'] ) );
 	}
 
 	/**
-	 * This test ensures that the `add_block_fields_to_schema()` method
-	 * works as expected.
+	 * This test ensures that the `register_interface_types()` method
+	 * works as expected when the get_allowed_block_types is used
 	 */
-	public function test_add_block_fields_to_schema() {
-		$post = 'Post';
+	public function test_add_block_fields_to_schema_with_get_allowed_block_types() {
+		add_filter(
+			'allowed_block_types_all',
+			function ( $allowed_blocks, $editor_context ) {
+				if ( isset( $editor_context->post ) && $editor_context->post instanceof \WP_Post && 'post' === $editor_context->post->post_type ) {
+					return array(
+						'core/image',
+						'core/paragraph',
+					);
+				}
+				return true;
```

Comment:
So if the filter API can either return a `string[]` of blocks, or `true` for all?